### PR TITLE
Flight phase works with millis_();

### PR DESCRIPTION
**Pull request by Joe-Joe-Joe-Joe**
- Updated can handler code to use ISR-safe freertos code and associated changes. Also fixed using an incorrect message getter
- Updated FreeRTOS daemon task priority so SetEventGroupBitsFromISR() will work as intended
- Fixed event bit comparison logic error
Tested with usb debug. Flight phase correctly transitions based on injector valve open message and delay parameters

### CODE REVIEW

**Comment by Joe-Joe-Joe-Joe** [comment]
There is a busyloop in there from testing. I need to double check if we still need that but probably not

File: STM32Cube/Core/Src/main.c [hunk 1]
```diff
@@ -188,25 +188,25 @@ int main(void)
...
+  //defaultTaskHandle = osThreadNew(StartDefaultTask, NULL, &defaultTask_attributes);
 
   /* USER CODE BEGIN RTOS_THREADS */
   /* add threads, ... */
   BaseType_t xReturned = pdPASS;
 
   //dunno if casting from CMSIS priorities is valid
   //xReturned &= xTaskCreate(vnIMUHandler, "VN Task", DEFAULT_STACKDEPTH_WORDS, NULL, (UBaseType_t) osPriorityNormal, &VNTaskHandle);
-  xReturned &= xTaskCreate(canHandlerTask, "CAN handler", DEFAULT_STACKDEPTH_WORDS, NULL, (UBaseType_t) osPriorityNormal, &canhandlerhandle);
+  //xReturned &= xTaskCreate(canHandlerTask, "CAN handler", DEFAULT_STACKDEPTH_WORDS, NULL, (UBaseType_t) osPriorityNormal, &canhandlerhandle);
```
Comment (Armaan-Sengupta) [inline STM32Cube/Core/Src/main.c:204]: Is this not needed, ig its not a big deal ppl can just uncomment what threads they need

File: STM32Cube/Core/Src/main.c [hunk 2]
```diff
@@ -219,6 +219,7 @@ int main(void)
 
   /* USER CODE BEGIN RTOS_EVENTS */
   /* add events, ... */
+  for(int i = 0; i <10000; i++){} //delay boot to allow millis_ timer to stabilize
```
Comment (Armaan-Sengupta) [inline STM32Cube/Core/Src/main.c:222]: Did this fix the "it doesnt work the first time the board is programmed"?

File: STM32Cube/Tasks/can_handler.c [hunk 3]
```diff
@@ -15,10 +16,14 @@ void can_handle_rx(const can_msg_t *message, uint32_t timestamp){
 	{
 		HAL_GPIO_WritePin(GPIOB, GPIO_PIN_5, 1);
 	}
-	else if(msgtype == MSG_ACTUATOR_CMD && get_actuator_id(message) == ACTUATOR_INJECTOR_VALVE && get_curr_actuator_state(message) == ACTUATOR_ON)
+	else if(msgtype == MSG_ACTUATOR_CMD && get_actuator_id(message) == ACTUATOR_INJECTOR_VALVE && get_req_actuator_state(message) == ACTUATOR_ON)
 	{
-		xEventGroupSetBits(flightPhaseEventsHandle, INJ_OPEN_BIT); //
+		result = xEventGroupSetBitsFromISR(flightPhaseEventsHandle, INJ_OPEN_BIT, &xHigherPriorityTaskWoken);
```
Comment (Armaan-Sengupta) [inline STM32Cube/Tasks/can_handler.c:21]: ISR safe version is based, I assume this function is called during an interrupt (sry idk can handler)?

**Review by Armaan-Sengupta — APPROVED** [review]
Looks good to me, @celery6 might wanna review some of the can handler ISR safe stuff, because I am not to familiar with can handler

File: STM32Cube/Core/Src/main.c (hunk 2 above)
Comment (Joe-Joe-Joe-Joe) [reply STM32Cube/Core/Src/main.c]: tested, I'm having trouble even replicating the issue? I've removed the line

**Pull request merged**